libvips/foreign/pdfiumload.c

```diff
@@ -584,47 +584,56 @@ vips_foreign_load_pdf_generate(VipsRegion *out_region,
 
 	top = r->top;
 	while (top < VIPS_RECT_BOTTOM(r)) {
-		VipsRect rect;
 		FPDF_BITMAP bitmap;
 
+		/* Is the rect within this page? It might not be if the output is more
+		 * than one tile wide and this page is narrower.
+		 */
+		VipsRect rect;
 		vips_rect_intersectrect(r, &pdf->pages[i], &rect);
+		if (rect.width > 0 &&
+			rect.height > 0) {
 
-		if (vips_foreign_load_pdf_get_page(pdf, pdf->page_no + i))
-			return -1;
+			if (vips_foreign_load_pdf_get_page(pdf, pdf->page_no + i))
+				return -1;
 
-		vips__worker_lock(&vips_pdfium_mutex);
+			vips__worker_lock(&vips_pdfium_mutex);
 
-		/* 4 means RGBA.
-		 */
-		bitmap = FPDFBitmap_CreateEx(rect.width, rect.height, 4,
-			VIPS_REGION_ADDR(out_region, rect.left, rect.top),
-			VIPS_REGION_LSKIP(out_region));
+			/* 4 means RGBA.
+			 */
+			if (!(bitmap = FPDFBitmap_CreateEx(rect.width, rect.height, 4,
+					VIPS_REGION_ADDR(out_region, rect.left, rect.top),
+					VIPS_REGION_LSKIP(out_region)))) {
+				g_mutex_unlock(&vips_pdfium_mutex);
```

Comment:
Can this really fail? If it can, we should set an error message, perhaps:

```suggestion
                vips_pdfium_error();
				g_mutex_unlock(&vips_pdfium_mutex);
```